### PR TITLE
first draft WIP device behavior updates

### PR DESCRIPTION
**Pull request by susanharper-okta**
## Description:
- **What's changed?** made updates to coincide with the device behavior updates being made
- **Is this PR related to a Monolith release?** Yes 2020.03.2 

### Resolves:

* [OKTA-283357](https://oktainc.atlassian.net/browse/OKTA-283357)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 1]
```diff
@@ -79,7 +79,57 @@ Okta will derive client request context directly from the HTTP request headers a
...
+
+`User-Agent: Mozilla/5.0 (<system-information>) <platform> (<platform-details>) <extensions>`
+
+Alternatively, you can make sure that the following string values are present in the User-Agent string so that the OS and browser are detected:
+
+**iOS**
+
+- The presence of the words `apple` or `ios` and at least one of these values: `iphone`, `ipad`, `ipod`, `ipad`
+
+  **Example:** `iPhone/ios/iPhone10,4/13.3.1/2020.2.6480/Owners iPhone`
```
Comment (rpamidimarri-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: Again, I think these examples are from a customer and we should remove them. We should just include the note about the presence of certain strings to make sure the OS and Browser show up correctly for IOS and Android.

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 2]
```diff
@@ -871,16 +871,28 @@ Content-Type: application/json
...
-You should send the device fingerprint only if the trusted app has a computed fingerprint for the end user's client.
+Specifying your own device fingerprint is a highly privileged operation that is limited to trusted web applications and requires making authentication requests with a valid API token. You should send the device fingerprint only if the trusted app has a computed fingerprint for the end user's client.
 
-**Notes:**
+> **Notes:** <br> Device fingerprint is different from the device token. The time and device based MFA in the Okta Sign-On policy rules depends on the device token only and not on the device fingerprint. To read more about the device token, see [Context Object](#context-object). The time and device based MFA would work only if you pass the device token in the [client request context](/docs/reference/api-overview/#client-request-context).<br><br> To use device fingerprinting for the new or unknown device email notification feature, include the `User-Agent` header in the request. For more information, see the [General Security documentation](https://help.okta.com/en/prod/Content/Topics/Security/Security_General.htm?).
```
Comment (rpamidimarri-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: > To use device fingerprinting for the new or unknown device email notification feature, include the `User-Agent` header in the request.

This is not accurate - we dont need the UA string for new device behavior to work.

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 3]
```diff
@@ -871,16 +871,28 @@ Content-Type: application/json
...
+* If you choose to build your own sign-in UX for a Native App, you must configure the App to use a trusted backend to pass a device token.
 
-* Device fingerprint is different from the device token. The time and device based MFA in Okta SignOn policy rules depends on the device token only and not on the device fingerprint. To read more about the device token, see [Context Object](#context-object). The time and device based MFA would work only if you send the device token passed in the [client request context](/docs/reference/api-overview/#client-request-context).
-* To use device fingerprinting for the new or unknown-device email notification feature, include the `User-Agent` header in the request. For more information, see the [General Security documentation](https://help.okta.com/en/prod/Content/Topics/Security/Security_General.htm?).
+> **Note:** Any change to the implementation will not match the Okta history of device fingerprints. All devices will look like **new** devices.
```
Comment (rpamidimarri-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: I think this note is not required - we put this note only for BOKF.

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 4]
```diff
@@ -871,16 +871,28 @@ Content-Type: application/json
...
+
+##### Device Fingerprint Best Practices
+
+Use the following recommendations as guidelines for generating and storing a device fingerprint for the `X-Device-Fingerprint` header
+for web apps and native apps.
+
+**Web Apps**<br>
+Okta recommends using a secure, HTTP-only cookie with a random/unique value on the customer's domain as the default implementation. See [Cookie flags that matter](https://odino.org/security-hardening-http-cookies/#cookie-flags-that-matter) for more best practices on hardening HTTP cookies.
+
+**Native Apps**<br>
```
Comment (rpamidimarri-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: @tthompson-okta FYI - since this is based on the notes you captured for BOKF. Can you please review this too?

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 5]
```diff
@@ -79,7 +79,57 @@ Okta will derive client request context directly from the HTTP request headers a
...
+
+Examples of User-Agent strings that can be correctly parsed by Okta:
+
+`Mozilla/5.0 (iPhone; CPU iPhone OS 13_3_1 like Mac OS X) AppleWebKit/534.46 (KHTML, like Gecko) Version/13.3.1 Mobile/9B179 Safari/7534.48.3 OktaMobile/6.3.0`
+
+`Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 10.0; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; .NET CLR 2.0.50727; .NET CLR 3.0.30729; .NET CLR 3.5.30729; Zoom 3.6.0)`
+
+`Mozilla/5.0 (Linux; Android 10; SM-G9600 Build/QP1A.190711.020; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/80.0.3987.119 Mobile Safari/537.36 OktaMobile/4.5.1`
+
+Examples of User-Agent strings that can't be correctly parsed by Okta:
```
[the remark this answers is not in the document]
Comment (rpamidimarri-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: fyi @srinivasanagandla-okta - since you helped investigate how the UA strings should be formatted for BOKF. Can you please review?

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 6]
```diff
@@ -79,7 +79,31 @@ Okta will derive client request context directly from the HTTP request headers a
...
+
+Make sure that the User-Agent string that your app constructs is in the correct format. This ensures that Okta can parse the OS and Browser fields. Good User-Agent strings that can be correctly parsed by Okta contain browser information, system information, platform, platform details, and any extensions.
+
+We recommend that you use a template like the following to format the User-Agent string:
+
+`User-Agent: Mozilla/5.0 (<system-information>) <platform> (<platform-details>) <extensions>`
+
+Alternatively, you can make sure that the following string values are present in the User-Agent string so that the OS and browser are detected:
+
+**Any Device Operating System**
```
Comment (srinivasanagandla-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: Since the following sentence is about `browser` I would recommend naming it as such. We could say something like `How to pass hint about the browser`.

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 7]
```diff
@@ -79,7 +79,31 @@ Okta will derive client request context directly from the HTTP request headers a
...
+
+If you have a browser string such as `chrome` or `safari`, add that to the User-Agent string.
+
+**iOS**
```
Comment (srinivasanagandla-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: Since the following `iOS` and `Android` is about the OS, we could group these two under the `How to pass OS information`

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 8]
```diff
@@ -79,7 +79,31 @@ Okta will derive client request context directly from the HTTP request headers a
...
+
+Include the words `apple` or `ios` and at least one of these values: `iphone`, `ipad`, `ipod`, `ipad`.
+
+**Android**
+
+Include the words `android` or `samsung`, which infers that Android is the operating system.
+
+> **Note:** Case doesn't matter since Okta converts the string to lowercase before matching.
```
Comment (srinivasanagandla-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: We could also add a note about testing to make sure that `OS` and `Browser` show up properly in the system log events.

Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: @srinivasanagandla-okta Can you help me out with the wording of this note?

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 9]
```diff
@@ -79,7 +79,34 @@ Okta will derive client request context directly from the HTTP request headers a
...
+
+**Pass a hint about the browser**
+
+If you have a browser string such as `chrome` or `safari`, add that to the User-Agent string.
+
+**Pass operating system information**
+
+- **iOS:** Include the words `apple` or `ios` and at least one of these values: `iphone`, `ipad`, `ipod`, `ipad`.
+- **Android:** Include the words `android` or `samsung`, which infers that Android is the operating system.
+
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: Can we add some examples here like ones specified in https://oktawiki.atlassian.net/wiki/spaces/eng/pages/900764066/Technical+Questions+from+Bank+Of+Oklahoma 
so users can see where they can be added in the string.

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 10]
```diff
@@ -79,7 +79,34 @@ Okta will derive client request context directly from the HTTP request headers a
...
+> **Note:** Okta recommends testing to make sure that Okta can parse both the `OS` and `Browser` fields from the User-Agent header that is passed by your native mobile app. See the [System Log](/docs/reference/api/system-log/#useragent-object).
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: How would the developer know how to test? if we include this Note we should also describe or point to info on how to test. For ex, "to test if Okta can parse the string, check the System log of the request that was passed. See the [System Log] ... "

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 11]
```diff
@@ -871,19 +871,27 @@ Content-Type: application/json
 
 Include the `X-Device-Fingerprint` header to supply a device fingerprint. The device fingerprint is used in the following ways:
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: to be consistent with product doc, "device fingerprint" should be "browser fingerprint". See https://oktawiki.atlassian.net/wiki/spaces/~damian.ali/pages/886806858/SIP+New+Device+Behavior for what's going into product help.

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 12]
```diff
@@ -871,19 +871,27 @@ Content-Type: application/json
...
 
-* If the new or unknown-device email notification is enabled, an email is sent to the user if the device fingerprint sent in the header is not associated with a previously successful user sign in. For more information about this feature, see the [General Security documentation](https://help.okta.com/en/prod/Content/Topics/Security/Security_General.htm?).
-* If you have the security behavior detection feature enabled and you have a new device behavior configured in a policy rule, a new device is detected if the device fingerprint sent in the header is not associated with previous successful logins. For more information about this feature, see [EA documentation](https://help.okta.com/en/prod/Content/Topics/Security/proc-security-behavior-detection.htm?).
+* If the new or unknown device email notification is enabled, an email is sent to the user if the device fingerprint sent in the header is not associated with a previously successful user sign in. For more information about this feature, see the [General Security documentation](https://help.okta.com/en/prod/Content/Topics/Security/Security_General.htm?).
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: can we replace "device fingerprint" here with "X-Device-Fingerprint" to be exact, since we're calling the header "browser fingerprint" when referring to it in plain English. (It's confusing because the header is mis-named. It's really a browser fingerprint and not device fingerprint"

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 13]
```diff
@@ -871,19 +871,27 @@ Content-Type: application/json
...
+* If you have the security behavior detection feature enabled and you have a new device behavior configured in a policy rule, a new device is detected if the device fingerprint sent in the header is not associated with a previously successful user sign in. For more information about this feature, see [EA documentation](https://help.okta.com/en/prod/Content/Topics/Security/proc-security-behavior-detection.htm?).
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: Same comment as the previous one, can we replace "device fingerprint" here with "X-Device-Fingerprint"

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 14]
```diff
@@ -871,19 +871,27 @@ Content-Type: application/json
...
 
-Specifying your own device fingerprint is a highly privileged operation limited to trusted web applications and requires making authentication requests with a valid API token.
-You should send the device fingerprint only if the trusted app has a computed fingerprint for the end user's client.
+Specifying your own device fingerprint is a highly privileged operation that is limited to trusted web applications and requires making authentication requests with a valid API token. You should send the device fingerprint only if the trusted app has a computed fingerprint for the end user's client.
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: replace "device fingerprint" with "browser fingerprint" for all instances here

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 15]
```diff
@@ -871,19 +871,27 @@ Content-Type: application/json
...
 
-**Notes:**
+> **Note:** Device fingerprint is different from the device token. The time and device based MFA in the Okta Sign-On policy rules depends on the device token only and not on the device fingerprint. To read more about the device token, see [Context Object](#context-object). The time and device based MFA would work only if you pass the device token in the [client request context](/docs/reference/api-overview/#client-request-context).
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: replace "device fingerprint" with "browser fingerprint" for all instances

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 16]
```diff
@@ -871,19 +871,27 @@ Content-Type: application/json
...
 
-* Device fingerprint is different from the device token. The time and device based MFA in Okta SignOn policy rules depends on the device token only and not on the device fingerprint. To read more about the device token, see [Context Object](#context-object). The time and device based MFA would work only if you send the device token passed in the [client request context](/docs/reference/api-overview/#client-request-context).
-* To use device fingerprinting for the new or unknown-device email notification feature, include the `User-Agent` header in the request. For more information, see the [General Security documentation](https://help.okta.com/en/prod/Content/Topics/Security/Security_General.htm?).
+##### Device Fingerprint Best Practices
 
-##### Request example for device fingerprinting
+Use the following recommendations as guidelines for generating and storing a device fingerprint for the `X-Device-Fingerprint` header for web apps and native apps.
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: replace "device fingerprint" with "browser fingerprint"

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 17]
```diff
@@ -871,19 +871,27 @@ Content-Type: application/json
...
+
+**Web Apps**<br>
+Okta recommends using a secure, HTTP-only cookie with a random/unique value on the customer's domain as the default implementation. See [Cookie flags that matter](https://odino.org/security-hardening-http-cookies/#cookie-flags-that-matter) for more best practices on hardening HTTP cookies.
 
+**Native Apps**<br>
+Ask the device operating system for a unique device ID. See [Apple's information on DeviceCheck](https://developer.apple.com/documentation/devicecheck) for an example.
+
+* Okta strongly encourages you to use the [AppAuth pattern](https://tools.ietf.org/html/rfc8252) (system browser) for Native App sign-in flows. When you use the AppAuth pattern, you can then leverage Okta's device token implementation without any custom code. This allows an untrusted application to still provide a proof-of-device attestation and leverage Okta's device behavior and per-device MFA policies. Also, it allows the App to support new factors and functionality without having to change code.
+* If you choose to build your own sign-in UX for a Native App, you must configure the App to use a trusted backend to pass a device token.
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: better wording: 
If you choose to build your own sign-in UI for a native application, the native application must use a trusted backend to pass the browser fingerprint to Okta. (would be great to refer to how make set a trusted backend here)

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 18]
```diff
@@ -871,19 +871,27 @@ Content-Type: application/json
...
+Use the following recommendations as guidelines for generating and storing a device fingerprint for the `X-Device-Fingerprint` header for web apps and native apps.
+
+**Web Apps**<br>
+Okta recommends using a secure, HTTP-only cookie with a random/unique value on the customer's domain as the default implementation. See [Cookie flags that matter](https://odino.org/security-hardening-http-cookies/#cookie-flags-that-matter) for more best practices on hardening HTTP cookies.
 
+**Native Apps**<br>
+Ask the device operating system for a unique device ID. See [Apple's information on DeviceCheck](https://developer.apple.com/documentation/devicecheck) for an example.
+
+* Okta strongly encourages you to use the [AppAuth pattern](https://tools.ietf.org/html/rfc8252) (system browser) for Native App sign-in flows. When you use the AppAuth pattern, you can then leverage Okta's device token implementation without any custom code. This allows an untrusted application to still provide a proof-of-device attestation and leverage Okta's device behavior and per-device MFA policies. Also, it allows the App to support new factors and functionality without having to change code.
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: - Replace "for Native App sign-in flows" with "for native application sign-ins". 
- @tthompson-okta  or @srinivasanagandla-okta  I think it's not clear to the developer what this means here "When you use the AppAuth pattern, you can then leverage Okta's device token implementation without any custom code. This allows an untrusted application to still provide a proof-of-device attestation and leverage Okta's device behavior and per-device MFA policies. Also, it allows the App to support new factors and functionality without having to change code." 
How is this related to device token and how is it relevant to browser fingerprint here? If we can't explain it easily we may want to take it out

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md (hunk 11 above)
Comment (rpamidimarri-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: I think we should continue to call the value for `X-Device-Fingerprint` header as a device fingerprint in the API docs. The documentation for browser flows in https://oktawiki.atlassian.net/wiki/spaces/~damian.ali/pages/886806858/SIP+New+Device+Behavior talks about the JS fingerprint that is generated by the Okta SIW, which can have FP and FN. For these trusted app API scenarios, the customer may be generating a UUID, storing in a secure cookie and passing it along in the header and in such cases, what they are sending is not the browser fingerprint. So, for the trusted app scenarios, I think it would be confusing if we call this browser fingerprint..

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md (hunk 1 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: @rpamidimarri-okta - @lilyhsiao-okta is requesting that we add some examples here. Can I get an iOS and an Android GENERIC example to include, please?

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md (hunk 8 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: @srinivasanagandla-okta - @lilyhsiao-okta has the following comment about the note. Can I get a little more information for this note? Such as a place to direct the user for more information on how to test? Here is Lily's comment:

How would the developer know how to test? if we include this Note we should also describe or point to info on how to test. For ex, "to test if Okta can parse the string, check the System log of the request that was passed. See the [System Log] ... "

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md (hunk 1 above)
Comment (srinivasanagandla-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: I would not recommend adding examples. The issue with examples is that people would literally copy and paste into their code which we do not want.

Comment (rpamidimarri-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: I agree with @srinivasanagandla-okta . There are so many ways a UA string can be constructed that we cannot show very generic examples and we dont want customers to copy the ones from the doc. The https://oktawiki.atlassian.net/wiki/spaces/eng/pages/900764066/Technical+Questions+from+Bank+Of+Oklahoma talks about specific examples based on the customer samples.. so, those examples make sense for that particular customer.

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md (hunk 18 above)
Comment (rpamidimarri-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: The AppAuth pattern (and the associated device token) would be applicable when we implement the new device behavior to also rely on the device token. Until then, I agree with @lilyhsiao-okta  that we should take this section out.

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md (hunk 17 above)
Comment (rpamidimarri-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: Inline with my comment above (https://github.com/okta/okta-developer-docs/pull/1049/files#r393387338), I think we should take this sentence out too.

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 19]
```diff
@@ -79,7 +79,34 @@ Okta will derive client request context directly from the HTTP request headers a
 
 Okta supports the standard `User-Agent` HTTP header to identify the user's browser or application. Always send a `User-Agent` string to uniquely identify your client application and version such as `Oktaprise/1.1`.
 
-> If your application is acting as a gateway or proxy, you should forward the `User-Agent` of the originating client with your API requests.
+> **Note:** If your application is acting as a gateway or proxy, you should forward the `User-Agent` of the originating client with your API requests.
+
+### Format a User-Agent string for native mobile applications
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: Just noticed that this says "for native mobile applications" 

It could just say "Format a User-Agent string" since the the paragraph below it is not specific to native mobile applications.

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md (hunk 1 above)
Comment (lilyhsiao-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: The reason for suggesting examples here is the sentence above it:

"you can make sure that certain string values are present in the User-Agent string so that the OS and Browser are detected" 

This makes it sound like the string values can be any where and it would be parsable. If that's not the case then we could re-word the sentence. 
That's where an example would be helpful.

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 20]
```diff
@@ -79,7 +79,57 @@ Okta will derive client request context directly from the HTTP request headers a
 
 Okta supports the standard `User-Agent` HTTP header to identify the user's browser or application. Always send a `User-Agent` string to uniquely identify your client application and version such as `Oktaprise/1.1`.
 
-> If your application is acting as a gateway or proxy, you should forward the `User-Agent` of the originating client with your API requests.
+> **Note:** If your application is acting as a gateway or proxy, you should forward the `User-Agent` of the originating client with your API requests.
+
+### Format a User-Agent string
```
[the remark this answers is not in the document]
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: @rpamidimarri-okta - shouldn't this remain generic since we aren't JUST talking about native mobile apps here?

[the remark this answers is not in the document]
Comment (rpamidimarri-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: yeah, makes sense

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 21]
```diff
@@ -77,9 +77,35 @@ Okta will derive client request context directly from the HTTP request headers a
...
+
+We recommend that you use a template like the following to format the User-Agent string:
+
+`User-Agent: Mozilla/5.0 (<system-information>) <platform> (<platform-details>) <extensions>`
+
+Okta recommends making test [authentication requests](/docs/reference/api/authn/#primary-authentication) and then checking for the related entries in the [System Log](/docs/reference/api/system-log/#useragent-object). Testing helps you make sure that Okta can parse both the `OS` and `Browser` fields from the `User-Agent` header that is passed by your application.
+
+If the `OS` and/or `Browser` fields come back as `Unknown` in the System Log, make sure that certain string values are present in the `User-Agent` string so that the OS and Browser are detected. For some Chrome examples, see [User-Agent strings](https://developer.chrome.com/multidevice/user-agent).
+
+> **Note:** Case doesn't matter since Okta converts the string to lowercase before matching.
```
Comment (rpamidimarri-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: we are not doing any matching here. I think you can get rid of this line.

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md [hunk 22]
```diff
@@ -77,9 +77,35 @@ Okta will derive client request context directly from the HTTP request headers a
...
+
+**Pass a hint about the browser**
+
+Add browser information such as `chrome` or `safari` to the `User-Agent` string.
+
+**Pass operating system information**
+
+- **iOS:** Include the words `apple` or `ios` and at least one of these values: `iphone`, `ipad`, `ipod`, `ipad`.
+
+- **Android:** Include the words `android` or `samsung`, which infers that Android is the operating system.
```
Comment (rpamidimarri-okta) [inline packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: I would remove `samsung` and just include `android`

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md (hunk 14 above)
Comment (srinivasanagandla-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: My take: I think we should keep the reference as "device fingerprint" for now to keep it consistent with the header name (X-Device-Fingerprint).

File: packages/@okta/vuepress-site/docs/reference/api-overview/index.md (hunk 22 above)
Comment (srinivasanagandla-okta) [reply packages/@okta/vuepress-site/docs/reference/api-overview/index.md]: sounds good to me.

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md (hunk 11 above)
Comment (lilyhsiao-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: Let's go with `X-device-Fingerprint` where it says "device fingerprint" here

Comment (lilyhsiao-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: Let's go with X-Device-Fingerprint where it says "device fingerprint" here

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md (hunk 15 above)
Comment (lilyhsiao-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: same as previous comment, let's just go with 'X-Device-Fingerprint' here

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 23]
```diff
@@ -871,19 +871,24 @@ Content-Type: application/json
...
-* To use device fingerprinting for the new or unknown-device email notification feature, include the `User-Agent` header in the request. For more information, see the [General Security documentation](https://help.okta.com/en/prod/Content/Topics/Security/Security_General.htm?).
+##### Device Fingerprint Best Practices
 
-##### Request example for device fingerprinting
+Use the following recommendations as guidelines for generating and storing a device fingerprint in the `X-Device-Fingerprint` header for both web and native applications.
+
+**Web Apps**<br>
+Okta recommends using a secure, HTTP-only cookie with a random/unique value on the customer's domain as the default implementation. See [Cookie flags that matter](https://odino.org/security-hardening-http-cookies/#cookie-flags-that-matter) for more best practices on hardening HTTP cookies.
```
Comment (lilyhsiao-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: 👍 on the link to cookie examples

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 24]
```diff
@@ -871,19 +871,24 @@ Content-Type: application/json
...
+* If the new or unknown device email notification is enabled, an email is sent to the user if the device fingerprint sent in the `X-Device-Fingerprint` header isn't associated with a previously successful user sign in. For more information about this feature, see the [General Security documentation](https://help.okta.com/en/prod/Content/Topics/Security/Security_General.htm?).
+* If you have the security behavior detection feature enabled and you have a new device behavior configured in a policy rule, a new device is detected if the device fingerprint sent in the `X-Device-Fingerprint` header isn't associated with a previously successful user sign in. For more information about this feature, see [EA documentation](https://help.okta.com/en/prod/Content/Topics/Security/proc-security-behavior-detection.htm?).
```
Comment (tthompson-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: Should we remove this bullet? This is the behavior we agreed is ultimately wrong and we are going to change, per the rollout plan here: https://oktawiki.atlassian.net/wiki/spaces/PM/pages/734406988/Fixing+Known+Device+behavior+for+V1+IDX#Fixing%22KnownDevice%22behaviorforV1&IDX-Rolloutplan

> Change doc to describe correct use of X-Device-Token (remove X-Device-Fingerprint from docs)

Although we haven't fixed the product behavior yet, I think we should either remove this or mark it with the `Deprecated` tag and explain that this use case is going to be superseded by passing Device Token instead (and link to the later section about Device Token).

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 25]
```diff
@@ -871,19 +871,24 @@ Content-Type: application/json
...
 
-Specifying your own device fingerprint is a highly privileged operation limited to trusted web applications and requires making authentication requests with a valid API token.
-You should send the device fingerprint only if the trusted app has a computed fingerprint for the end user's client.
+Specifying your own device fingerprint in the `X-Device-Fingerprint` header is a highly privileged operation that is limited to trusted web applications and requires making authentication requests with a valid API token. You should send the device fingerprint only if the trusted app has a computed fingerprint for the end user's client.
 
-**Notes:**
+> **Note:** Device fingerprint is different from the device token. The time and device based MFA in the Okta Sign-On policy rules depends on the device token only and not on the device fingerprint. To read more about the device token, see [Context Object](#context-object). The time and device based MFA would work only if you pass the device token in the [client request context](/docs/reference/api-overview/#client-request-context).
```
Comment (tthompson-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: > time and device based MFA

The "time" portion of this doesn't sound right. Why would time-based MFA have any relation to device fingerprint? I thought time-based MFA was based on your existing session, which keeps a timestamp of "lastMFAChallenge".

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md (hunk 24 above)
Comment (lilyhsiao-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: If a customer enabled New device behavior detection though they would need to pass this header in. So I think we may need to wait until we have the new solution before we can Deprecate this.

File: packages/@okta/vuepress-site/docs/reference/api/authn/index.md [hunk 26]
```diff
@@ -869,21 +869,26 @@ Content-Type: application/json
...
-**Notes:**
+> **Note:** The `X-Device-Fingerprint` header is different from the device token. Device-based MFA in the Okta Sign-On policy rules depends on the device token only and not on the `X-Device-Fingerprint` header. To read more about the device token, see [Context Object](#context-object). Device-based MFA would work only if you pass the device token in the [client request context](/docs/reference/api-overview/#client-request-context).
 
-* Device fingerprint is different from the device token. The time and device based MFA in Okta SignOn policy rules depends on the device token only and not on the device fingerprint. To read more about the device token, see [Context Object](#context-object). The time and device based MFA would work only if you send the device token passed in the [client request context](/docs/reference/api-overview/#client-request-context).
-* To use device fingerprinting for the new or unknown-device email notification feature, include the `User-Agent` header in the request. For more information, see the [General Security documentation](https://help.okta.com/en/prod/Content/Topics/Security/Security_General.htm?).
+##### Device Fingerprint Best Practices
```
Comment (tthompson-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: This is confusing, we are adding the best practices for Device *Token* and saying it's the best practices for *Fingerprint*.

We should re-work this whole section to better describe the top-level use cases and then how they map to implementation details like Device Token vs Device Fingerprint. (I know we have some forthcoming changes to make the product work correctly, but we need to make the docs correct ahead of time and explain what's changing). I thought PM was going to help structure this, but here's how I thought it would look:

| Use Case | Features Available | Notes |
| --------- | ------------------- | ------ |
| Require MFA for users on unknown devices |  | Encourage admins to configure default (fall-through) rules to be secure (MFA) by default, and configure per-device rules to *opt-out* of MFA, not opt-in. |
| | Per-device option in App SignOn Policy | Based on Device Token |
| | Risk Score in Okta SignOn Policy | Based weighted combo of Device Token + Device Fingerprint + other signals |
| | New Device Behavior in Okta SignOn Policy (only for advanced mode; we should [discourage folks from using](https://oktawiki.atlassian.net/wiki/spaces/PM/pages/925597862/Rethinking+IP+based+security+behaviors) unless they have an advanced case that Risk Score doesn't address) | Based on Device Fingerprint. NOTE: Okta has determined that Device Fingerprint is not a strong proof of device. This feature will change to use Device Token in an upcoming release. |
| Send "New Device" notifications to users | New Sign In Notification Email (under Security->General) | Based on Device Fingerprint |

https://oktawiki.atlassian.net/wiki/spaces/PM/pages/734406988?focusedCommentId=903167566#comment-903167566

To this end, "Primary Authentication with Device Fingerprinting" really shouldn't be a first-class scenario like we're making it:

Authentication

It should be "Primary Authentication with Known Device", and then we map this back to Use Case 1 in that table above.

Comment (rpamidimarri-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authn/index.md]: > We should re-work this whole section to better describe the top-level use cases and then how they map to implementation details like Device Token vs Device Fingerprint. (I know we have some forthcoming changes to make the product work correctly, but we need to make the docs correct ahead of time and explain what's changing)

I think it would be confusing to document forthcoming product changes (particularly while we are considering how to standardize header vs context body). In my opinion, the use cases and the notes you captured, while valid, should be documented when we [finalize the plan - I will setup the design meeting next week](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/916015568/Technical+Design+for+Improvements+to+New+Device+Behavior+Detection).

> This is confusing, we are adding the best practices for Device Token and saying it's the best practices for Fingerprint.

I think this is basically providing the same best practices that we did to the customer couple of weeks back. That customer was also sending a UUID through the X-Device-Fingerprint. Until we implement the proposal we were discussing to improve the behavior, we dont want customers to generate browser fingerprints using AuthJS and we want them to send UUIDs (that are securely stored in cookies). To that end, I think the way the doc is written out now is correct - It captures the best way to use the feature today. We will have to come back and update the doc again when we finalize/implement the proposal.
